app/controllers/publications/monthly_statistics_controller.rb

```diff
@@ -29,16 +26,21 @@ def calculate_download_sizes(report)
       end.compact
     end
 
-    def temporarily_unavailable; end
-
   private
 
     def current_report
-      if params[:month].present?
-        MonthlyStatisticsTimetable.current_report_at(Date.parse("#{params[:month]}-01"))
+      return MonthlyStatisticsTimetable.report_for_current_period if params[:month].blank?
+
+      if itt_match.present?
```

Comment:
how about having a separate route:
```
get '/monthly-statistics/ITT(:year)' => 'monthly_statistics#show', as: :monthly_report_itt
```
and then this condition is `if params[:year].present?` and we don't need to bother with regex matching here.

Reply:
and this action becomes a bit simpler:

```
    def current_report
      if params[:month].present?
        MonthlyStatisticsTimetable.current_report_at(Date.parse("#{params[:month]}-01"))
      elsif params[:year].present?
        MonthlyStatisticsTimetable.report_for_latest_in_cycle(params[:year].to_i)
      else
        MonthlyStatisticsTimetable.report_for_current_period
      end
    end
```

Reply:
Done ♻️